functable.c

```diff
@@ -237,6 +237,10 @@ static void init_functable(void) {
 #endif
 
     // Assign function pointers individually for atomic operation
+#if defined(_MSC_VER)
+#  pragma warning( push )
+#  pragma warning( disable: 4054)
```

Comment:
I can't find any documentation on 4054...

Reply:
Me neither... I tried really hard to search MSDN, but it seems to be literally undocumented online.

As far as I know, the warning is about casting function pointer to `void *`, which is allowed by default, but loses the prototype of the underlying function, so isn't really safe.

Reply:
Even though we are casting it to `(void *)` in `FUNCTABLE_ASSIGN` it is still giving the warning?

Reply:
Some compiler versions are giving the warning, but not all... I don't have all the toolchain versions for all the supported Visual C++ versions and platforms, so I can't test it locally (yet). Basically it's the reverse of C4055, which is obsolete since VC++ 2017 but still documented.

Reply:
Would using `&VAR.FUNC_NAME` be better? I noticed we do it in the `FUNCTABLE_ASSIGN` for clang but not MSVC?

Reply:
> Would using `&(VAR.FUNC_NAME)` be better? I noticed we do it in the `FUNCTABLE_ASSIGN` for clang but not MSVC?

That would return the address of the functable entry, not the pointer to the function stored in functable entry. It really depends on the prototype of the atomic swap function which one is correct.